pkg/cmd/repo/delete/delete.go

```diff
@@ -30,18 +31,23 @@ func NewCmdDelete(f *cmdutil.Factory, runF func(*DeleteOptions) error) *cobra.Co
 	}
 
 	cmd := &cobra.Command{
-		Use:   "delete <repository>",
+		Use:   "delete [<repository>]",
 		Short: "Delete a repository",
 		Long: `Delete a GitHub repository.
 
+With no argument, deletes the current repository. Otherwise, deletes the specified repository.
+
 Deletion requires authorization with the "delete_repo" scope. 
 To authorize, run "gh auth refresh -s delete_repo"`,
-		Args: cmdutil.ExactArgs(1, "cannot delete: repository argument required"),
+		Args: cobra.MaximumNArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			opts.RepoArg = args[0]
+			if len(args) > 0 {
+				opts.RepoArg = args[0]
+			}
 			if !opts.IO.CanPrompt() && !opts.Confirmed {
 				return cmdutil.FlagErrorf("could not prompt: confirmation with prompt or --confirm flag required")
```

Comment:
I know this was not introduced in this PR but what do you think about
```suggestion
				return cmdutil.FlagErrorf("--confirm required when not running interactively")
```